maat-scheduled-tasks/src/main/java/uk/gov/justice/laa/maat/scheduled/tasks/service/TrialDataService.java

```diff
@@ -28,32 +30,46 @@ public class TrialDataService {
 
     private final StoredProcedureService storedProcedureService;
 
-    public void populateTrialData() {
-        log.info("Starting to populate Trial Data in to Hub.");
+    @Transactional
```

Comment:
I think we might get some inconsistencies having the transactional annotation at this high level. If my understanding is correct then if there is an issue with say the stored procedure call which raises a type of Runtime Exception then this will trigger a rollback of the database changes. However the S3 changes won't get rolled back.

Reply:
Had another look through the flow and I can see that the flow should be fine now, as like you say we are only marking the files as complete at the end and any files marked as errored earlier won't be written to the DB or have any other action against them.